src/Github/CreateReleaseTextViaKeepAChangelog.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\AutomaticReleases\Github;
+
+use Laminas\AutomaticReleases\Git\Value\BranchName;
+use Laminas\AutomaticReleases\Git\Value\SemVerVersion;
+use Laminas\AutomaticReleases\Github\Api\GraphQL\Query\GetMilestoneChangelog\Response\Milestone;
+use Laminas\AutomaticReleases\Github\Value\RepositoryName;
+use Phly\KeepAChangelog\Common\ChangelogParser;
+use Phly\KeepAChangelog\Exception\ExceptionInterface;
+
+use function file_exists;
+use function file_get_contents;
+
+class CreateReleaseTextViaKeepAChangelog implements CreateReleaseText
+{
+    public function __invoke(
+        Milestone $milestone,
+        RepositoryName $repositoryName,
+        SemVerVersion $semVerVersion,
+        BranchName $sourceBranch,
+        string $repositoryDirectory
+    ): string {
+        /** @psalm-var non-empty-string $changelog */
```

Comment:
This is not guaranteed by the `ChangelogParser`: let's instead use `Assert::notEmpty()` **after** `$changelog` has been assigned a `string`

Reply:
:+1:

That said, `ChangelogParser` will raise an exception if no changelog is found and/or it is empty (it's looking for non-empty contents, IIRC). But the assertion makes it bullet-proof.